### DISCUSSION TITLE
Manually call model.eval() and model.train() inside the training loop

### DISCUSSION THREAD

**Discussion by rwbfd** [Lightning-AI/pytorch-lightning#6718] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
How to manually call `model.eval()` or `model.train()` inside the lightning module? I happen to have several models and not all of them need to be updated during each forward pass. Thanks!

**Comment by kkirtac** [comment, marked as answer]
You can use [`self.freeze()`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py#L1479) and [`self.unfreeze()`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py#L1494).

**Reply by rwbfd** [reply]
Hi thanks for the reply. However, what if I have two models and I need to freeze one while unfreeze the other one?

**Reply by kkirtac** [reply]
That shouldn't make a difference. Here in this [example](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py#L101) there are 2 models, i.e., encoder and decoder. If your encoder and decoder models are type of `nn.Module` lets say, you can call anyone like `self.encoder.train()` or `self.encoder.eval()` and the same applies for the `self.decoder` as well. If they are type of `pl.LightningModule` then you can use freeze() and unfreeze() as examplified above.
